From 65056c7bae4d30f6918799abe4bfaa4bb8aba8e2 Mon Sep 17 00:00:00 2001
From: Break Yang <breakds@gmail.com>
Date: Thu, 15 Aug 2024 08:59:12 +0800
Subject: [PATCH] Fix Popen call

---
 aim/storage/migrations/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aim/storage/migrations/utils.py b/aim/storage/migrations/utils.py
index d4a53195..5843b1d4 100644
--- a/aim/storage/migrations/utils.py
+++ b/aim/storage/migrations/utils.py
@@ -13,7 +13,7 @@ def create_migration_cmd():
         migrations_config = os.path.join(storage_dir, 'migrations', 'alembic.ini')
     else:
         migrations_config = os.path.join(storage_dir, 'migrations', 'alembic_dev.ini')
-    return [sys.executable, '-m', 'alembic', '-c', migrations_config, 'upgrade', 'head']
+    return ['alembic', '-c', migrations_config, 'upgrade', 'head']
 
 
 def upgrade_database(db_url):
-- 
2.44.1

