From df78988851200a6c9318503c2258a81d18e2aeab Mon Sep 17 00:00:00 2001
From: Break Yang <breakds@gmail.com>
Date: Thu, 1 Aug 2024 00:08:13 -0700
Subject: [PATCH] Fix

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 50683dc..c798b10 100644
--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@ except ImportError:
 
 
 aimrocks_extra_compile_args = [
-    '-std=c++11',
+    '-std=c++17',
     '-O3',
     '-Wall',
     '-Wextra',
-- 
2.44.1

