From 7cf8a3b801cc9a64b7fba3e14ce0684cb5364ca4 Mon Sep 17 00:00:00 2001
From: Break Yang <breakds@gmail.com>
Date: Sun, 16 Jul 2023 14:14:30 -0700
Subject: [PATCH] [Patch] remove hatch-pypi

---
 pyproject.toml | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 58fc0a11..e6bf1b2a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ['hatchling', 'hatch-fancy-pypi-readme>=22.5.0']
+requires = ['hatchling']
 build-backend = 'hatchling.build'
 
 [tool.hatch.version]
@@ -128,22 +128,6 @@ Funding = 'https://github.com/sponsors/samuelcolvin'
 Source = 'https://github.com/pydantic/pydantic'
 Changelog = 'https://docs.pydantic.dev/latest/changelog/'
 
-[tool.hatch.metadata.hooks.fancy-pypi-readme]
-content-type = 'text/markdown'
-# construct the PyPI readme from README.md and HISTORY.md
-fragments = [
-    {path = "README.md"},
-    {text = "\n## Changelog\n\n"},
-    {path = "HISTORY.md", pattern = "(.+?)<!-- package description limit -->"},
-    {text = "\n... see [here](https://docs.pydantic.dev/changelog/#v0322-2019-08-17) for earlier changes.\n"},
-]
-# convert GitHuB issue/PR numbers and handles to links
-substitutions = [
-    {pattern = '(\s+)#(\d+)', replacement = '\1[#\2](https://github.com/pydantic/pydantic/issues/\2)'},
-    {pattern = '(\s+)@([\w\-]+)', replacement = '\1[@\2](https://github.com/\2)'},
-    {pattern = '@@', replacement = '@'},
-]
-
 [tool.pytest.ini_options]
 testpaths = 'tests'
 xfail_strict = true
-- 
2.40.1

