diff --git a/setup.py b/setup.py
index 8ece515..47ea4bd 100644
--- a/setup.py
+++ b/setup.py
@@ -101,7 +101,7 @@ class BuildMJBindingsCommand(setuptools.Command):
         '--header_paths={}'.format(self.header_paths),
         '--output_dir={}'.format(output_dir)
     ]
-    self.announce('Running command: {}'.format(command), level=logging.DEBUG)
+    # self.announce('Running command: {}'.format(command), level=logging.DEBUG)
     try:
       # Prepend the current directory to $PYTHONPATH so that internal imports
       # in `autowrap` can succeed before we've installed anything.
