From fc3d281351643e41c9a1da46653dfd910206434f Mon Sep 17 00:00:00 2001
From: Break Yang <breakds@gmail.com>
Date: Tue, 14 Dec 2021 13:28:18 -0800
Subject: [PATCH] Fix setup.py

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 1c12366..47c5088 100644
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ version = "0.2.5.1"
 
 install_requires = [
     "gym",
-    "numpy<=1.19.3",
+    "numpy",
     "matplotlib",
     "pandas",
     "pygame",
@@ -40,7 +40,6 @@ install_requires = [
     "panda3d-simplepbr",
     "pillow",
     "pytest",
-    "opencv-python-headless",
 ]
 
 # if (not is_mac()) and (not is_win()):
-- 
2.36.0

